File: practice15/8.cpp
//https://www.nowcoder.com/practice/b90f1f5957ae4ee09b9dddf464718587?tpId=3&tqId=10934&tPage=4&rp=&ru=/ta/hackathon&qru=/ta/hackathon/question-ranking
#include<stdio.h>
#include<iostream>
using namespace std;
int main()
{
	int i = -1, j = -1, n, r = 1;
	while (cin >> n)
	{
		r = 1;
		i = -1, j = -1;
		int a[100][100] = { 0 };
		while (r <= n*n)
		{
			for (i++, j++; a[i][j] == 0 && j<n; j++)
			{
				a[i][j] = r++;
			}
			for (i++, j--; a[i][j] == 0 && i<n; i++)
			{
				a[i][j] = r++;
			}
			for (i--, j--; a[i][j] == 0 && j >= 0; j--)
			{
				a[i][j] = r++;
			}
			for (i--, j++; a[i][j] == 0 && i>0; i--)
			{
				a[i][j] = r++;
			}
		}
		for (i = 0; i < n; i++)
		{
			for (j = 0; j < n; j++)
			{
				cout << a[i][j] << " ";
			}
			cout << endl;
		}
		cout << endl;
	}
	system("pause");
	return 0;
}

File: practice31/8.java



//https://www.nowcoder.com/practice/16e48900851646c0b2c6cdef9d7ea051?tpId=90&tqId=30878&tPage=6&rp=6&ru=/ta/2018test&qru=/ta/2018test/question-ranking
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        String[] str=bufferedReader.readLine().split(" ");
        int m=Integer.parseInt(str[0]);
        int n=Integer.parseInt(str[1]);
        List<Integer> pt_m=to_two(m);
        List<Integer> pt_n=to_two(n);
        int count=0;
        int x;
        if(pt_m.size()>pt_n.size()){
            for(int i=pt_n.size();i<pt_m.size();i++){
                x=pt_m.get(i);
                if(pt_m.get(i)==1){
                    count++;
                }
            }
            for(int i=0,j=0;j<pt_n.size();i++,j++){
                if(pt_m.get(i)!=pt_n.get(j)){
                    count++;
                }
            }
        }
        else if(pt_m.size()<pt_n.size()){
            for(int i=pt_m.size();i<pt_n.size();i++){
                if(pt_n.get(i)==1){
                    count++;
                }
            }
            for(int i=0,j=0;i<pt_m.size();i++,j++){
                if(pt_m.get(i)!=pt_n.get(j)){
                    count++;
                }
            }
        }
        else{
            for(int i=0,j=0;i<pt_m.size();i++,j--){
                if(pt_m.get(i)!=pt_n.get(j)){
                    count++;
                }
            }
        }
        System.out.println(count);
    }
    static List to_two(int n){
        List<Integer> result=new ArrayList<>();
        while(n>0){
            result.add(n%2);
            n=n/2;
        }
        return result;
    }
}